### PR TITLE
Add release notes

### PR DESCRIPTION
**Pull request by gassmoeller**
Preliminary release notes for 3.0.0. Do not merge yet! 
I open this now so @geodynamics/aspect-developers can take a look to:
 - check if I forgot major changes
 - correct mistakes in author attribution or description of features
 - typos or rewording suggestions are welcome. 
 
This is only the list of the most relevant changes that will be included in the announcement emails and forum post, the full list of changes will be available on the website.

### CODE REVIEW

File: contrib/release/release-tasklist.md [hunk 1]
```diff
@@ -198,6 +198,118 @@ and the links are working
 
 ## List of prior release notes
 
+Announcement for 3.0.0 (Nov 6, 2024)
+-----------------------------------------
```
Comment (bangerth) [inline contrib/release/release-tasklist.md:202]: I think you want to move the announcement for 3.0.0 *above* the list of prior release notes? Or is this file only for us, and we keep this announcement here as a starting point for our own release process?

**Review by bangerth — COMMENTED** [review]
The list looks good to me!

**Review by tjhei — APPROVED** [review]
Very nice. It looks like we have quite a few new big features!

File: contrib/release/release-tasklist.md (hunk 1 above)
Comment (gassmoeller) [reply contrib/release/release-tasklist.md:202]: This file is mostly for us and internal use to keep track of all official announcements. We can make it more prominent in the future, but then I think all of this needs to go into a different file (or files). I will keep it in place for now.

**Comment by gassmoeller** [comment]
Let's merge this and use it as the release notes. I think this is a good time to create the release branch, keep it around for a week, and only allow bugfixes to go in.

**Pull request merged**